### PR TITLE
Fix ids on upgrades

### LINKED ISSUE #795: Room entries for upgraded rooms retain their old room id for mapping
This makes things like unprovisioning unsuccessful.

### PR DESCRIPTION
**Pull request by Half-Shot**
This is the IRC bridge side of https://github.com/matrix-org/matrix-appservice-bridge/pull/129

Fixes #795 

**EDIT**: This depends on ~~https://github.com/matrix-org/matrix-appservice-bridge/pull/132~~, we are being bitten by a bug.

**EDIT 2**: This also depends on ~~#803~~

### CODE REVIEW

File: lib/bridge/IrcBridge.js [hunk 1]
```diff
@@ -20,7 +20,8 @@ var BridgeRequest = require("../models/BridgeRequest");
 var stats = require("../config/stats");
 var DataStore = require("../DataStore");
 var log = require("../logging").get("IrcBridge");
-const { Bridge, MatrixUser, PrometheusMetrics, Logging } = require("matrix-appservice-bridge");
+const { Bridge, MatrixUser, MatrixRoom,
```
Comment (anoadragon453) [inline lib/bridge/IrcBridge.js]:
```suggestion
const {
    Bridge,
    MatrixUser,
    MatrixRoom,
```

nit

File: lib/bridge/IrcBridge.js [hunk 2]
```diff
@@ -20,7 +20,8 @@ var BridgeRequest = require("../models/BridgeRequest");
...
+    PrometheusMetrics, Logging } = require("matrix-appservice-bridge");
```
Comment (anoadragon453) [inline lib/bridge/IrcBridge.js]:
```suggestion
    PrometheusMetrics,
    Logging,
} = require("matrix-appservice-bridge");
```

/me guesses if this is how it works in JS land.

File: lib/bridge/IrcBridge.js [hunk 3]
```diff
@@ -1048,6 +1050,19 @@ IrcBridge.prototype._fetchJoinedRooms = Promise.coroutine(function*() {
     }
 });
 
+
+IrcBridge.prototype._roomUpgradeMigrateEntry = function(entry, newRoomId) {
```
Comment (anoadragon453) [inline lib/bridge/IrcBridge.js]: nit: Might be nice to have a comment here along the lines of https://github.com/matrix-org/matrix-appservice-bridge/pull/129 for why this function is necessary.

File: lib/bridge/IrcBridge.js (hunk 2 above)
Comment (Half-Shot) [reply lib/bridge/IrcBridge.js]: yis

**Pull request merged**